src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/HeartbeatDefaultPayload.cs

```diff
@@ -3,19 +3,58 @@
     using System;
     using System.Collections.Generic;
     using System.Linq;
+    using System.Net.Http;
     using System.Reflection;
+    using System.Threading.Tasks;
 
     internal static class HeartbeatDefaultPayload
     {
-       public static readonly string[] DefaultFields =
-        {
+        public static readonly string[] DefaultFields =
+         {
             "runtimeFramework",
-            "baseSdkTargetFramework"
+            "baseSdkTargetFramework",
+            "osVersion"
+        };
+
+        public static readonly string[] DefaultOptionalFields =
+         {
+            "processId",
+            // the  following are Azure Instance Metadata fields
+            "osType",
+            "location",
+            "name",
+            "offer",
+            "platformFaultDomain",
+            "platformUpdateDomain",
+            "publisher",
+            "sku",
+            "version",
+            "vmId",
+            "vmSize"
         };
 
-        public static void PopulateDefaultPayload(IEnumerable<string> disabledFields, IHeartbeatProvider provider)
+        public static readonly string[] AllDefaultFields = DefaultFields.Union(DefaultOptionalFields).ToArray();
+
+        /// <summary>
+        /// Azure Instance Metadata Service exists on a single non-routable IP on machines configured
+        /// by the Azure Resource Manager. See <a href="https://go.microsoft.com/fwlink/?linkid=864683">to learn more.</a>
+        /// </summary>
+        private static string baseImdsUrl = $"http://169.254.169.254/metadata/instance/compute";
```

Comment:
Yes, that is a definite requirement here (and appears to be one of the reasons my build is failing perhaps?). Thanks again for the link to an example! I think I simply have to ensure my calls are wrapped by the internal SDK call enter- and exit- here.

Reply:
Call `SdkInternalOperation.Start/Stop` around this call to address this feedback

Reply:
@SergeyKanzhelev ```SdkInternalOperation.Start/Stop``` is only to prevent FirstChanceExceptionmodule from reporting ex right? I was referring to the DependencyTelemetry item collected by DepCollector module. The change would be in web repository in the file linked above.

Reply:
It is already implemented https://github.com/Microsoft/ApplicationInsights-dotnet-server/blob/develop/Src/DependencyCollector/Shared/Implementation/ApplicationInsightsUrlFilter.cs#L59